File: translation-library/python/examples/basic_usage.py
"""
Примеры базового использования библиотеки перевода
"""

from translation_library import TranslationLibrary, TranslatorProvider


def basic_google_translation():
    """Базовый перевод с использованием Google Translate"""
    print("=== Базовый перевод Google Translate ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.GOOGLE,
    )
    
    result = library.translate(
        text='Hello, world!',
        target_language='ru',
    )
    
    print(f'Исходный текст: Hello, world!')
    print(f'Перевод: {result.text}')
    print(f'Провайдер: {result.provider}')
    print()


def basic_yandex_translation():
    """Базовый перевод с использованием Yandex Translate"""
    print("=== Базовый перевод Yandex Translate ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.YANDEX,
        # API ключ можно передать здесь или через переменную окружения
        # default_api_key='your-yandex-api-key',
    )
    
    result = library.translate(
        text='Hello, world!',
        target_language='ru',
    )
    
    print(f'Исходный текст: Hello, world!')
    print(f'Перевод: {result.text}')
    print(f'Провайдер: {result.provider}')
    print()


def basic_libretranslate_translation():
    """Базовый перевод с использованием LibreTranslate"""
    print("=== Базовый перевод LibreTranslate ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.LIBRETRANSLATE,
        default_endpoint='https://libretranslate.de/translate',
    )
    
    result = library.translate(
        text='Hello, world!',
        target_language='ru',
    )
    
    print(f'Исходный текст: Hello, world!')
    print(f'Перевод: {result.text}')
    print(f'Провайдер: {result.provider}')
    print()


def translation_with_source_language():
    """Перевод с указанием исходного языка"""
    print("=== Перевод с указанием исходного языка ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.GOOGLE,
    )
    
    result = library.translate(
        text='Hello, world!',
        target_language='ru',
        source_language='en',
    )
    
    print(f'Исходный текст: Hello, world!')
    print(f'Исходный язык: en')
    print(f'Перевод: {result.text}')
    print()


def translation_with_options():
    """Перевод с использованием TranslationOptions"""
    print("=== Перевод с использованием TranslationOptions ===")
    
    from translation_library import TranslationOptions
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.GOOGLE,
    )
    
    options = TranslationOptions(
        target_language='de',
        source_language='en',
    )
    
    result = library.translate(
        text='Hello, world!',
        target_language=options.target_language,
        source_language=options.source_language,
    )
    
    print(f'Исходный текст: Hello, world!')
    print(f'Перевод: {result.text}')
    print()


def detect_language():
    """Определение языка текста"""
    print("=== Определение языка текста ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.GOOGLE,
    )
    
    test_texts = [
        'Hello, world!',
        'Привет, мир!',
        'Bonjour le monde!',
        'Hallo Welt!',
    ]
    
    for text in test_texts:
        lang = library.detect_language(text)
        print(f'Текст: {text} -> Язык: {lang}')
    print()


def get_supported_languages():
    """Получение списка поддерживаемых языков"""
    print("=== Список поддерживаемых языков (первые 10) ===")
    
    library = TranslationLibrary(
        default_provider=TranslatorProvider.GOOGLE,
    )
    
    languages = library.get_supported_languages()
    
    print(f'Всего поддерживаемых языков: {len(languages)}')
    print(f'Первые 10 языков: {languages[:10]}')
    print()


if __name__ == '__main__':
    basic_google_translation()
    # basic_yandex_translation()  # Требует API ключ
    # basic_libretranslate_translation()  # Может быть медленным
    translation_with_source_language()
    translation_with_options()
    detect_language()
    get_supported_languages()

File: translation-library/nodejs/examples/basic_usage.ts
/**
 * Примеры базового использования библиотеки перевода
 */

import { TranslationLibrary, TranslatorProvider } from '../index';

/**
 * Базовый перевод с использованием Google Translate
 */
async function basicGoogleTranslation(): Promise<void> {
  console.log('=== Базовый перевод Google Translate ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: 'ru',
  });

  console.log(`Исходный текст: Hello, world!`);
  console.log(`Перевод: ${result.text}`);
  console.log(`Провайдер: ${result.provider}`);
  console.log('');
}

/**
 * Базовый перевод с использованием Yandex Translate
 */
async function basicYandexTranslation(): Promise<void> {
  console.log('=== Базовый перевод Yandex Translate ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.YANDEX,
    // API ключ можно передать здесь или через переменную окружения
    // defaultApiKey: 'your-yandex-api-key',
  });

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: 'ru',
  });

  console.log(`Исходный текст: Hello, world!`);
  console.log(`Перевод: ${result.text}`);
  console.log(`Провайдер: ${result.provider}`);
  console.log('');
}

/**
 * Базовый перевод с использованием LibreTranslate
 */
async function basicLibretranslateTranslation(): Promise<void> {
  console.log('=== Базовый перевод LibreTranslate ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.LIBRETRANSLATE,
    defaultEndpoint: 'https://libretranslate.de/translate',
  });

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: 'ru',
  });

  console.log(`Исходный текст: Hello, world!`);
  console.log(`Перевод: ${result.text}`);
  console.log(`Провайдер: ${result.provider}`);
  console.log('');
}

/**
 * Перевод с указанием исходного языка
 */
async function translationWithSourceLanguage(): Promise<void> {
  console.log('=== Перевод с указанием исходного языка ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: 'ru',
    sourceLanguage: 'en',
  });

  console.log(`Исходный текст: Hello, world!`);
  console.log(`Исходный язык: en`);
  console.log(`Перевод: ${result.text}`);
  console.log('');
}

/**
 * Перевод с использованием TranslationOptions
 */
async function translationWithOptions(): Promise<void> {
  console.log('=== Перевод с использованием TranslationOptions ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const options = {
    targetLanguage: 'de',
    sourceLanguage: 'en',
  };

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: options.targetLanguage,
    sourceLanguage: options.sourceLanguage,
  });

  console.log(`Исходный текст: Hello, world!`);
  console.log(`Перевод: ${result.text}`);
  console.log('');
}

/**
 * Определение языка текста
 */
async function detectLanguage(): Promise<void> {
  console.log('=== Определение языка текста ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const testTexts = [
    'Hello, world!',
    'Привет, мир!',
    'Bonjour le monde!',
    'Hallo Welt!',
  ];

  for (const text of testTexts) {
    const lang = await library.detectLanguage(text);
    console.log(`Текст: ${text} -> Язык: ${lang}`);
  }
  console.log('');
}

/**
 * Получение списка поддерживаемых языков
 */
async function getSupportedLanguages(): Promise<void> {
  console.log('=== Список поддерживаемых языков (первые 10) ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const languages = await library.getSupportedLanguages();

  console.log(`Всего поддерживаемых языков: ${languages.length}`);
  console.log(`Первые 10 языков: ${languages.slice(0, 10).join(', ')}`);
  console.log('');
}

/**
 * Обработка ошибок
 */
async function errorHandling(): Promise<void> {
  console.log('=== Обработка ошибок ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  try {
    // Пустой текст вызовет ошибку
    await library.translate({
      text: '',
      targetLanguage: 'ru',
    });
  } catch (error) {
    if (error instanceof Error) {
      console.log(`Ошибка перевода: ${error.message}`);
    }
  }
  console.log('');
}

/**
 * Настройка таймаута и количества попыток
 */
async function customTimeoutAndRetries(): Promise<void> {
  console.log('=== Настройка таймаута и количества попыток ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
    defaultTimeout: 60000, // 60 секунд
    defaultRetries: 5,
  });

  const result = await library.translate({
    text: 'Hello, world!',
    targetLanguage: 'ru',
  });

  console.log(`Перевод: ${result.text}`);
  console.log('');
}

/**
 * Перевод с сохранением ссылок
 */
async function translationWithLinks(): Promise<void> {
  console.log('=== Перевод с сохранением ссылок ===');

  const library = new TranslationLibrary({
    defaultProvider: TranslatorProvider.GOOGLE,
  });

  const textWithLinks = `
    Привет! Посетите наш сайт [https://example.com](https://example.com)
    Или посмотрите [документацию](https://docs.example.com)
  `;

  const result = await library.translate({
    text: textWithLinks,
    targetLanguage: 'en',
  });

  console.log(`Исходный текст: ${textWithLinks}`);
  console.log(`Перевод: ${result.text}`);
  console.log('');
}

async function main(): Promise<void> {
  await basicGoogleTranslation();
  // await basicYandexTranslation(); // Требует API ключ
  // await basicLibretranslateTranslation(); // Может быть медленным
  await translationWithSourceLanguage();
  await translationWithOptions();
  await detectLanguage();
  await getSupportedLanguages();
  await errorHandling();
  await customTimeoutAndRetries();
  await translationWithLinks();
}

main().catch(console.error);